### ISSUE TITLE
Tile caching behavior in OL@10

### ISSUE THREAD

**Issue by itspangler** [bplmaps/atlascope-v2#71]
At the border of Swampscott and Lynn, two neighborhoring atlases—both Hopkins 1880—load at the same time:

<img width="1436" alt="Screenshot 2024-11-29 at 11 11 27 AM" src="https://github.com/user-attachments/assets/bb2d7663-99ef-466f-a986-a7e3f4d9fb99">

This is a rare bug occasioned by the fact that two different atlases by the same publisher and from the same year appear within the same view. Need to look at the `volumes` file and figure out how to ensure they don't both appear when one is selected.

**Comment by garrettdashnelson** [comment]
At a first glance, I think this is due to tile caching behavior and not to any issue with the volumes file. My first guess is that OL@10 must have introduced some new caching functions for tile layers and we probably have to wipe the cache every time we swap the source of a layer.

**Comment by itspangler** [comment]
oh, i think you're right - it happens when switching any layer... maybe we can troubleshoot next week.

**Comment by itspangler** [comment]
possibly the `clear()` method detailed here? flagging so i can look later

"Remove all cached reprojected tiles from the source. The next render cycle will create new tiles."

https://openlayers.org/en/latest/apidoc/module-ol_source_XYZ-XYZ.html

**Comment by garrettdashnelson** [comment]
@itspangler Just a note to say that I tried quite a few variants of flushing the cache with the `.clear()` method (it's technically on a `TileJSON` source, not an `XYZ` source) without any luck. My suspicion is that even if the network cache is cleared, the new OL is probably not unpainting the canvas and leaving existing tiles painted onto the canvas until they are overpainted by something else. It's possible we might have to paint out the canvas with black first, then clear cache, then switch to new TileJSON source.

**Comment by itspangler** [comment]
@garrettdashnelson were you able to replicate this error locally? i made a branch for `caching-bug-fix`, but i couldn't replicate the behavior that i am seeing in production. also not able to replicate on `staging`

**Comment by garrettdashnelson** [comment]
Yes, I'm able to reproduce it locally. `staging` is not up-to-date with the latest changes to `main`, so it's still not running OL@10. (I can fix that now.) You might check whether your branch is coming off of the head of `main` or not.

**Comment by mradamcox** [comment]
FWIW I think I was experiencing the same thing on OldInsuranceMaps.net with the georeferencing preview, swapping the source used to work but after an upgrade to OL 10 it started to leave the artifacts. I solved it by just creating a new layer with the new source, adding it to the map, and removing the old layer (there was no real need for me to swap sources on the same layer).

**Comment by garrettdashnelson** [comment]
Closed by #79
